### PR TITLE
Refs #153 Upgrade jQuery to 1.12.4

### PR DESCRIPTION
**Pull request by joehybird**
Upgrade to last compatible jQuery version with our codebase and do some cleanup work to prepare migration to 2.x

TODO : Changelog

### CODE REVIEW

**Review by genglert — CHANGES_REQUESTED** [review]

File: creme/creme_core/static/creme_core/js/lib/browser.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /*******************************************************************************
     Creme is a free/open-source Customer Relationship Management software
-    Copyright (C) 2020  Hybird
+    Copyright (C) 2021  Hybird
```
[the remark this answers is not in the document]
Comment (joehybird) [reply creme/creme_core/static/creme_core/js/lib/browser.js:3]: Ah! My pre-push checks aren't working perfectly -_-'

File: CHANGELOG.txt [hunk 2]
```diff
@@ -9,6 +9,7 @@
   Users side :
   ------------
     # The version of Django has been upgraded to "3.1".
+    # The version of jQuery has been upgraded to "1.12.4"
```
Comment (genglert) [inline CHANGELOG.txt:12]: You can remove this one (django is in users' section because of installation/virtual env)

File: CHANGELOG.txt [hunk 3]
```diff
@@ -17,13 +18,35 @@
...
+    - The version of jQuery is now "1.12.4"
 
 
     Non breaking changes :
     ----------------------
         # Deprecations :
             - The class 'creme_core.forms.fields.AjaxModelChoiceField' is deprecated.
             - In the template "creme_core/generics/blockform/field-block.html", the block {% block extra_buttons %} is deprecated.
+            - In Javascript:
+                * Some creme.ajax.* functions are now deprecated; Use creme.ajax.Query instead:
```
Comment (genglert) [inline CHANGELOG.txt:30]: s/Use/use

File: CHANGELOG.txt [hunk 4]
```diff
@@ -17,13 +18,35 @@
...
+                    - creme.ajax.reloadContent()
+                    - creme.ajax.json.send()
+                    - creme.ajax.json.get()
+                    - creme.ajax.json.post()
+                    - creme.ajax.json.isvalid()
+                    - creme.ajax.json.parse()
+                * Some creme.form.* functions are now deprecated
+                    - creme.forms.Select.optionsFromData()
+                    - creme.forms.Select.fill()
+                * All 'creme.layout.Layout' classes are deprecated; Use CSS instead
```
Comment (genglert) [inline CHANGELOG.txt:44]: s/Use/use

**Review by genglert — CHANGES_REQUESTED** [review]

**Review by genglert — APPROVED** [review]

**Pull request merged**